### ISSUE TITLE
Expose seed at top-level synthesize() for reproducibility

### ISSUE THREAD

**Issue by MohamedEldushi** [dataxid/dataxid-python#90]
# Expose seed at top-level synthesize() for reproducibility

## Type
**Feature Request** — add a `seed` keyword argument to `dataxid.synthesize()` (and consistently across `Model.create()` / `model.generate()`) so users can obtain reproducible synthetic data without constructing a `ModelConfig` manually.

## Summary

Today, the only way to get reproducible synthetic data from DataXID is:

```python
import dataxid
cfg = dataxid.ModelConfig(seed=42)
model = dataxid.Model.create(data=df, config=cfg)
synthetic = model.generate(n_samples=100)
```

This:
1. Breaks the "5 lines of code" promise in the Quickstart.
2. Forces users to discover `ModelConfig` (14/16 parameters of which are undocumented — see #79).
3. Forces users to switch from `synthesize()` to `Model.create()` + `generate()` just to set a single parameter.

A top-level `seed` argument fixes all three:

```python
synthetic = dataxid.synthesize(data=df, n_samples=100, seed=42)
```

### 1. Reproducibility is not optional for DataXID's target market

The PDF's own introduction positions DataXID as **privacy-preserving synthetic data** for use cases where privacy laws apply. Those same regulated contexts — healthcare, finance, audit — require reproducibility: given the same input and the same seed, the same synthetic output must be derivable months later for validation, reproduction of studies, or legal defense.

A synthetic-data tool without an obvious `seed` parameter signals to regulated users that the tool wasn't designed with their workflow in mind.

### 2. Current shortcut is invisible to Quickstart readers

`ModelConfig(seed=None)` exists today but is undocumented (#79, parameter 15 of 16). Even a careful user reading the PDF and Quickstart will not know it exists. Users who google "dataxid reproducible" will not find it.

### 3. Industry convention — every serious library does this

| Library | Top-level way to seed |
|---|---|
| `numpy` | `np.random.default_rng(42)` |
| `pandas` | `df.sample(n=100, random_state=42)` |
| `sklearn` | `train_test_split(..., random_state=42)`, `KMeans(random_state=42)`, etc. |
| `torch` | `torch.manual_seed(42)` |
| `xgboost` | `XGBClassifier(random_state=42)` |
| `tensorflow` | `tf.random.set_seed(42)` |
| **`dataxid`** | ❌ **no top-level option** |

DataXID is the odd one out.

### 4. Enables better bug reports and better CI tests

In preparing the companion issues (#77, #78, #79), I had to seed manually via `ModelConfig(seed=42)` so that DataXID engineers reviewing the bug reports could reproduce my exact findings. A top-level `seed` would have removed friction from that workflow — and will do the same for every future user filing an issue.

## Proposed API

```python
# Single-call (the Quickstart path)
synthetic = dataxid.synthesize(data=df, n_samples=1000, seed=42)

# Split-call path
model = dataxid.Model.create(data=df, seed=42)       # seed on Model.create too
synthetic = model.generate(n_samples=1000, seed=7)   # optional: reseed generation separately

# Existing path continues to work
cfg = dataxid.ModelConfig(seed=42)
model = dataxid.Model.create(data=df, config=cfg)
```

`seed` should accept:

- `None` (default) — current behavior (non-deterministic)
- `int` — standard 64-bit seed, deterministic output
- `numpy.random.Generator` (optional future) — pre-seeded RNG for advanced users

## Suggested Implementation

Minimal change — `synthesize()` already builds a `ModelConfig` internally when one isn't provided. Just thread the keyword through:

```python
# dataxid/__init__.py  (conceptual)

def synthesize(
    data,
    n_samples,
    *,
    seed: int | None = None,
    verbose: bool = False,          # paired with feature #<F7>
    config: ModelConfig | None = None,
):
    if config is None:
        config = ModelConfig(seed=seed)
    else:
        if seed is not None and config.seed is not None and config.seed != seed:
            raise ValueError(
                "Both `seed=` and `config.seed=` were set to different values. "
                "Pass seed via only one of them."
            )
        if seed is not None:
            config = replace(config, seed=seed)   # dataclasses.replace

    model = Model.create(data=data, config=config, verbose=verbose)
    try:
        return model.generate(n_samples=n_samples)
    finally:
        model.delete()
```

## Benefits Summary

| Benefit | Impact |
|---|---|
| Quickstart stays 5 lines, now with reproducibility | Low-friction onboarding |
| Matches every major library's convention | Removes the "unfinished" impression |
| Enables regulated / audited deployments | Opens healthcare & finance markets |
| Makes bug reports reproducible by default | Maintainers get clean repros |
| Enables deterministic CI tests | Users can assert `synthetic.equals(expected)` |
| Implementation cost | ~10 lines, one file, non-breaking |

## Alternatives Considered

1. **Keep `ModelConfig(seed=...)` as the only path, but document it.** Helps, but still forces users to learn about `ModelConfig` for a single common knob. Not enough.
2. **Expose every `ModelConfig` parameter as a kwarg on `synthesize()`.** Overkill — adds 16 parameters and a maintenance burden. `seed` is special because it's the one parameter every serious user will want.
3. **Global state** (`dataxid.set_seed(42)`). Anti-pattern — breaks composability, thread safety, and testing. Do not recommend.

## Related Reports

- **#79 (undocumented ModelConfig surface)** — this feature makes one of the 16 undocumented parameters a first-class citizen. The other 15 can stay on `ModelConfig`; only `seed` needs to be top-level.
- **Feature #<F7> (`verbose=True`)** — natural sibling kwarg. Both follow the same "elevate the most useful knob to top-level" pattern.

## Acceptance Criteria

1. `dataxid.synthesize(data=df, n_samples=100, seed=42)` accepts `seed` without error.
2. Two successive calls with the same seed produce identical output (byte-for-byte, or at minimum statistically identical within numerical tolerance).
3. Two calls with different seeds produce different output.
4. Setting `seed` in both `synthesize(seed=...)` and `config=ModelConfig(seed=...)` with conflicting values raises a clear `ValueError`.
5. Quickstart in the docs is updated to show the reproducible form.

## Multi-Environment Confirmation

Re-run on Google Colab on 2026-04-21:

- **Platform:** Linux-6.6.113+ (Colab)
- **Python:** 3.12.13
- **dataxid:** **0.3.0**

On 0.3.0, `dataxid.ModelConfig` still exposes `seed` (default `None`) but `dataxid.synthesize()` does **not** accept a top-level `seed` keyword argument. Users wanting reproducibility must still construct a `ModelConfig(seed=...)` and switch to the `Model.create()` / `generate()` flow, exactly as described in this feature request.

Signature inspected on Colab:
```python
>>> import inspect, dataxid
>>> print(inspect.signature(dataxid.synthesize))
# seed is not among the parameters
```

**Comment by BurakUlver** [comment]
This is in 0.3.1. You can pass seed on synthesize() and generate() now:

```python
synthetic = dataxid.synthesize(data=df, n_samples=100, seed=42)
```
